src/main/kotlin/com/sourcegraph/cody/autocomplete/action/CodyActionPromoter.kt

```diff
@@ -0,0 +1,29 @@
+package com.sourcegraph.cody.autocomplete.action
+
+import com.intellij.openapi.actionSystem.ActionPromoter
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.DataContext
+import com.intellij.openapi.editor.actionSystem.EditorAction
+
+class CodyActionPromoter : ActionPromoter {
+  override fun promote(
+      actions: MutableList<out AnAction>,
+      context: DataContext
+  ): MutableList<AnAction>? {
+    return if (actions.stream().noneMatch { action: AnAction? ->
```

Comment:
nit: it would be nice to extract a variable out here
```suggestion
    val isCodyAction = actions.stream().noneMatch { action: AnAction? -> ...
    return if (isCodyAction) { null } else {
```